### PR TITLE
Fix `Scrollbar` can be dragged  from above/below the main axis on desktop.

### LINKED ISSUE #107765: ScrollBar moves with an offset if cursor leaves flutter application.
This issue is reproducible on both the stable and master branch, and affects both desktop and web applications. I checked the behaviour with web apps such as YouTube and the chrome browser and it seems that they take into consideration the position of the cursor when deciding as to whether to move the ScrollBar.

<details>

<summary>video of bug in Flutter:</summary>

https://user-images.githubusercontent.com/73116038/179329223-c0bf6738-3cfa-4f8f-bcdc-6a88b5e9e6bb.mp4

</details>

<details>

<summary>video of expected behaviour in YouTube:</summary>

https://user-images.githubusercontent.com/73116038/179329228-da40bd35-8cb1-43fb-80b3-fe756663769d.mp4

</details>

<details>

<summary>Reproducible code:</summary>

    import 'package:flutter/material.dart';
    
    void main() => runApp(const MyApp());
    
    class MyApp extends StatelessWidget {
      const MyApp({Key? key}) : super(key: key);
    
      static const String _title = 'Flutter Code Sample';
    
      @override
      Widget build(BuildContext context) {
        return MaterialApp(
          title: _title,
          home: Scaffold(
            appBar: AppBar(title: const Text(_title)),
            body: const Center(
              child: MyStatefulWidget(),
            ),
          ),
        );
      }
    }
    
    class MyStatefulWidget extends StatefulWidget {
      const MyStatefulWidget({Key? key}) : super(key: key);
    
      @override
      State<MyStatefulWidget> createState() => _MyStatefulWidgetState();
    }
    
    class _MyStatefulWidgetState extends State<MyStatefulWidget> {
      final ScrollController _firstController = ScrollController();
    
      @override
      Widget build(BuildContext context) {
        return Scrollbar(
          thumbVisibility: true,
          controller: _firstController,
          child: ListView.builder(
              controller: _firstController,
              itemCount: 100,
              itemBuilder: (BuildContext context, int index) {
                return Padding(
                  padding: const EdgeInsets.all(8.0),
                  child: Text('Scrollable 1 : Index $index'),
                );
              }),
        );
      }
    }

</details>

<details>

<summary>flutter doctor -v</summary>

    [✓] Flutter (Channel master, 3.1.0-0.0.pre.1633, on LMDE 5 (elsie)
        5.10.0-16-amd64, locale en_GB.UTF-8)
        • Flutter version 3.1.0-0.0.pre.1633 on channel master at
          /home/henryr/Dev/SDK/Flutter/flutter
        • Upstream repository https://github.com/flutter/flutter.git
        • Framework revision 95a5a8d741 (2 days ago), 2022-07-13 09:53:40 -0700
        • Engine revision 20038216ac
        • Dart version 2.18.0 (build 2.18.0-272.0.dev)
        • DevTools version 2.15.0
    
    [✓] Android toolchain - develop for Android devices (Android SDK version
        32.1.0-rc1)
        • Android SDK at /home/henryr/Android/Sdk
        • Platform android-32, build-tools 32.1.0-rc1
        • Java binary at:
          /home/henryr/Dev/SDK/AndroidStudio/android-studio/jre/bin/java
        • Java version OpenJDK Runtime Environment (build
          11.0.12+0-b1504.28-7817840)
        • All Android licenses accepted.
    
    [✓] Chrome - develop for the web
        • Chrome at google-chrome
    
    [✓] Linux toolchain - develop for Linux desktop
        • Debian clang version 11.0.1-2
        • cmake version 3.18.4
        • ninja version 1.10.1
        • pkg-config version 0.29.2
    
    [✓] Android Studio (version 2021.2)
        • Android Studio at /home/henryr/Dev/SDK/AndroidStudio/android-studio
        • Flutter plugin version 69.0.2
        • Dart plugin version 212.5744
        • Java version OpenJDK Runtime Environment (build
          11.0.12+0-b1504.28-7817840)
    
    [✓] Connected device (2 available)
        • Linux (desktop) • linux  • linux-x64      • LMDE 5 (elsie) 5.10.0-16-amd64
        • Chrome (web)    • chrome • web-javascript • Google Chrome 103.0.5060.114
    
    [✓] HTTP Host Availability
        • All required HTTP hosts are available
    
    • No issues found!

</details>

### PR DESCRIPTION
**Pull request by TahaTesser**
fixes https://github.com/flutter/flutter/issues/107765

This PR fixes the issue where `Scrollbar` can be dragged above or below the viewport on the desktop.

All the desktop platforms seem to have the same scroll behavior, (Scrollbar can only be dragged the dragging parallel to the cross axis).
https://github.com/flutter/flutter/issues/107765 has reproduced the issue [Linux](https://github.com/flutter/flutter/issues/107765#issue-1306662763) and [Windows](https://github.com/flutter/flutter/issues/107765#issuecomment-1186866431) compared it with native applications and I have tested native macOS applications. 

### Bug

https://user-images.githubusercontent.com/48603081/180462185-25a264a0-b28b-4a3d-a4ae-0f2f2fd11b34.mov

### Fix

https://user-images.githubusercontent.com/48603081/180462220-ba87e6e1-19db-4021-833a-b72e9de6ed35.mov

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [ ] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

File: packages/flutter/lib/src/widgets/scrollbar.dart [hunk 1]
```diff
@@ -1634,16 +1634,16 @@ class RawScrollbarState<T extends RawScrollbar> extends State<T> with TickerProv
     late double primaryDelta;
     switch (position.axisDirection) {
       case AxisDirection.up:
-        primaryDelta = _dragScrollbarAxisOffset!.dy - updatedOffset.dy;
+        primaryDelta = clampDouble(_dragScrollbarAxisOffset!.dy, 0, position.viewportDimension) - updatedOffset.dy;
```
Comment (werainkhatri) [inline packages/flutter/lib/src/widgets/scrollbar.dart:1637]: I don't think `clamp`ing is the correct or expected behavior, especially in MacOS (haven't tested in Windows or Linux). As per the video below, which is Wikipedia open in Safari on MacOS Monoterey 12.4, looks like the last position where the mouse was "clicking" just before the scrollbar was about to overflow is saved and scrolling can resume only when the mouse crosses the last position.


https://user-images.githubusercontent.com/44755140/180601757-2aab6011-7a26-48ee-8038-fb4e2d2eeffc.mov

Comment (TahaTesser) [reply packages/flutter/lib/src/widgets/scrollbar.dart:1637]: Good catch, I will look into it.

Comment (bernaferrari) [reply packages/flutter/lib/src/widgets/scrollbar.dart:1637]: macOS with trackpad behaves differently

Comment (werainkhatri) [reply packages/flutter/lib/src/widgets/scrollbar.dart:1637]: looks same to me, mouse or touchpad. do you mind sharing a video?

Comment (TahaTesser) [reply packages/flutter/lib/src/widgets/scrollbar.dart:1637]: My colleague @huycozy tested this on all three platforms (Windows, macOS, and Linux) and concluded it's the same behavior on the other two platforms as @werainkhatri pointed out, thanks! 


Based on these new findings, I will figure out the new scrolling scrollbar behavior.

**Comment by TahaTesser** [comment]
Currently, updated offset and drag offset are not sufficient to distinguish drag from a specific location.
I have other priorities before my vacation but I will come to this after.

**Pull request closed without merging**